lib/puppet/parser/environment_compiler.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/parser/compiler'
+
+class Puppet::Parser::EnvironmentCompiler < Puppet::Parser::Compiler
+  def self.compile(env)
+    $env_module_directories = nil
+    env.check_for_reparse
+
+    node = Puppet::Node.new(env)
+    node.environment = env
+    new(node).compile
+  rescue => detail
+    message = "#{detail} in environment #{env.name}"
+    Puppet.log_exception(detail, message)
+    raise Puppet::Error, message, detail.backtrace
+  end
+
+  def compile
+    Puppet.override(@context_overrides, "For compiling environment #{environment.name}") do
+      @catalog.environment_instance = environment
+
+      Puppet::Util::Profiler.profile("Compile: Created settings scope", [:compiler, :create_settings_scope]) { create_settings_scope }
+
+      activate_binder
+
+      Puppet::Util::Profiler.profile("Compile: Evaluated main", [:compiler, :evaluate_main]) { evaluate_main }
+
+      Puppet::Util::Profiler.profile("Compile: Evaluated application instances", [:compiler, :evaluate_applications]) { evaluate_applications }
+
+      Puppet::Util::Profiler.profile("Compile: Finished catalog", [:compiler, :finish_catalog]) { finish }
```

Comment:
These profiler identifiers, should they be different than the corresponding ones in the regular compiler? Say use "Env Compile" ?